### PR TITLE
Fixes NSObjectInaccessibleException

### LINKED ISSUE #436: Crash: NSFetchedResultsController + NSObjectInaccessibleException
#### Scenario:
1. Simperium is deleting entities in background.
2. At the exact same time, there is a NSFetchedResultsController instance running a NSFetchRequest.
3. Due to the `resultsController performFetch:` inner workings, objects are initially loaded as faults.
4. During the FRC's section calculation, the objects will be loaded into memory.

#### Results:
There is a (really hard to trigger) race condition between steps (1) and (3), in which the objects what were loaded as faults are no longer relevant.

Problem in this scenario is that the NSFetchedResultsController instance is locking the main thread, and thus, preventing the NSManagedObjectContextDidSaveNotification from getting merged. Which, in turn, causes invalid objects to be accessed.

#### Notes:
This is a side effect of not using NSManagedObjectContext's parenting mechanism, which has been removed due to a Core Data Deadlock (Reference Issue #376).

### PR DESCRIPTION
**Pull request by jleandroperez**
Closes #436

/cc @astralbodies (*THANKS* in advance!!!)

### CODE REVIEW

File: Simperium/SPCoreDataStorage.m [hunk 1]
```diff
@@ -485,14 +492,45 @@ - (void)mainContextObjectsDidChange:(NSNotification *)notification {
...
+    // NOTE:
+    // Deleting an entity in a Children MOC, while there was a faulted reference in the MainMOC, might trigger a NSObjectInaccessibleException (#436).
+    // Workaround: we'll make sure the objects are actually loaded into the WriterMOC. This will effectively prevent an exception,
+    // and the object will get removed as soon as the DidSave note is merged.
+
+    [self.writerManagedObjectContext performBlockAndWait:^{
+        for (NSManagedObjectID *objectID in workerDeletedIds) {
+            NSManagedObject *writerMO = [self.writerManagedObjectContext existingObjectWithID:objectID error:nil];
+            if (writerMO.isFault) {
+                [writerMO willAccessValueForKey:nil];
```
Comment (astralbodies) [inline Simperium/SPCoreDataStorage.m]: This does seem dirty but if it works its better than a crash. :smile: I can't see this being a problem if Apple ever fixes it either.

**Comment by astralbodies** [comment]
:shipit:

**Comment by jleandroperez** [comment]
@astralbodies **Thank you sir**!!!

File: Simperium/SPCoreDataStorage.m (hunk 1 above)
Comment (jleandroperez) [inline Simperium/SPCoreDataStorage.m]: Agreed, i'm not quite happy either. But... same thought as you... better than a crash!

**Pull request merged**